src/controller/students.controller.ts

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable prefer-const */
+/* eslint-disable import/no-unresolved */
+import express, { Request, Response } from 'express';
+import { IStudent } from '../interface/student.interface';
+import { ITacher } from '../interface/tacher.interface';
+import StudentSchema from '../models/student.model';
+import TacherSchema from '../models/tacher.model';
+import * as valdiat from './tacher.controller';
+
+const router = express.Router();
+
+// Add Student
+export const postStudent = async (req: Request, res: Response) => {
+  const student = new StudentSchema({
+    // get all body Parameters
+    sName: req.body.sName,
+    age: req.body.age,
+    averageGrade: req.body.averageGrade,
+    studentID: req.body.studentID,
+  });
+
+  // Save the new student
+  try {
+    const newStudent = await student.save();
+    res.status(201).json(newStudent);
+  } catch (err: any) {
+    res.status(400).json({ message: err.message });
+  }
+};
+
+// Print one students by ID
+export const getStudent = async (req: Request, res: Response) => {
+  try {
+    const student: IStudent = await StudentSchema.findOne({
+      studentID: req.query.studentID,
+    });
+    res.send(student);
+  } catch (err: any) {
+    res.status(500).json({ message: err.message });
+  }
+};
+
+// Edit student
+export const editStudent = async (req: Request, res: Response) => {
+  try {
+    const student: IStudent = await StudentSchema.updateOne(
+      { studentID: req.query.studentID },
+      {
+        sName: req.body.sName,
+        age: req.body.age,
+        averageGrade: req.body.averageGrade,
+      }
+    );
+    res.status(201).json(student);
+  } catch (err: any) {
+    res.status(400).json({ message: err.message });
+  }
+};
+
+// Delete student
+export const delStudent = async (req: Request, res: Response) => {
+  try {
+    valdiat.delStudentFromTacher(req.query.studentID);
+    const student: IStudent = await StudentSchema.deleteOne({
+      studentID: req.query.studentID,
+    });
+    res.status(200).send(student);
+  } catch (err: any) {
+    res.status(500).json({ message: err.message });
+  }
+};
+
+// Get student profession (no Query Params here)
+export const getStudentProfession = async (req: Request, res: Response) => {
+  try {
+    // Aggregate the students profession from tachers collection
+    const student: IStudent[] = await StudentSchema.aggregate([
```

Comment:
same as before. this aggregation should be in the repository

Reply:
done